src/actions/index.js

```diff
@@ -33,13 +34,41 @@ export const userLogout = (_, callback) => {
   }
 }
 
+export const guestUserLogin = (data, callback) => {
+  return async dispatch => {
+    try {
+      const res = await axios.get(getGuestUserLoginUrlApi(), data)
+      dispatch({ type: 'LOG_IN', payload: res.data, isGuestAuth: true })
+      callback(res.data.status)
+    } catch (err) {
+      callback(err.response.data.errors.nonFieldErrors[0])
+    }
+  }
+}
+
 export const whoami = () => {
   return async (dispatch, getState) => {
     try {
       const res = await axios.get(getWhoAmIApi())
+
+      const roles = res.data.roles
+      
+      var isGuest =false;
+
+      for (var index = 0; index < roles.length; ++index) {
```

Comment:
You should use an already written function in the `formula_one` for the same purpose: https://omniport.readthedocs.io/en/latest/references/formula_one/functions/if_role.html#if-role